app/controllers/chat.js

```
@@ -0,0 +1,60 @@
+
+    static EVENTS = {
+        NEW_MESSAGE: 'new_message',
+        SYSTEM_MESSAGE_CHUNK: 'system_message_chunk',
+        SYSTEM_MESSAGE_END: 'system_message_end',
+        MESSAGE_END: 'message_end',
+        DISCONNECT: 'disconnect',
+        ERROR: 'error',
+    }
+
+    async initialize(socket) {
+        const { characterId } = socket?.handshake?.query;
```

Biome

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)